### PR TITLE
fix(add): Better error message when missing npm specifier

### PR DESCRIPTION
**Pull request by nathanwhit**
Before:
Screenshot 2024-08-09 at 3 15 01 PM

After:
Screenshot 2024-08-09 at 3 52 15 PM

### CODE REVIEW

**Comment by marvinhagemeister** [comment]
Nice, this is a great DX improvement! Wondering if we should we print `deno add npm:ajv` without the version as the command.

File: cli/tools/registry/pm.rs [hunk 1]
```diff
@@ -178,6 +178,7 @@ fn package_json_dependency_entry(
 pub async fn add(
   flags: Arc<Flags>,
   add_flags: AddFlags,
+  cmd_name: &'static str,
```
Comment (bartlomieju) [inline cli/tools/registry/pm.rs:181]: Can you use an enum here?

**Comment by nathanwhit** [comment]
> Nice, this is a great DX improvement! Wondering if we should we print `deno add npm:ajv` without the version as the command as that would match more what the user would type. I'm mainly worried that otherwise users would think that they'd need to provide the version arg themselves.

Good idea! I changed it so that we suggest whatever the user specified, but with `npm:`. So if you do
```
deno add ajv
```
we suggest 
```
deno add npm:ajv
```
and if you do
```
deno add ajv@8
```
we suggest
```
deno add npm:ajv@8
```

**Comment by marvinhagemeister** [comment]
> > Nice, this is a great DX improvement! Wondering if we should we print `deno add npm:ajv` without the version as the command as that would match more what the user would type. I'm mainly worried that otherwise users would think that they'd need to provide the version arg themselves.
> 
> Good idea! I changed it so that we suggest whatever the user specified, but with `npm:`. So if you do
> 
> ```
> deno add ajv
> ```
> 
> we suggest
> 
> ```
> deno add npm:ajv
> ```
> 
> and if you do
> 
> ```
> deno add ajv@8
> ```
> 
> we suggest
> 
> ```
> deno add npm:ajv@8
> ```

Perfect, this is even better!

**Review by marvinhagemeister — APPROVED** [review]

**Pull request merged**